python/src/rtree.rs

```diff
@@ -277,6 +309,76 @@ impl RTree {
         result
     }
 
+    #[classmethod]
+    #[pyo3(
+        signature = (buf),
+        text_signature = "(buf)")
+    ]
+    #[allow(clippy::too_many_arguments)]
+    pub fn from_buffer(
+        _cls: &PyType,
+        py: Python,
+        buf: PyObject,
+    ) -> PyResult<Self> {
+        buf.extract::<RTree>(py)
+    }
+    #[pyo3(
+        signature = (),
+        text_signature = "()")
+    ]
+    #[allow(clippy::too_many_arguments)]
+    pub fn to_buffer<'py>(
+        &'py self,
+        py: Python<'py>,
+    ) -> PyObject {
+        PyBytes::new_bound(py, self.0.buffer()).into()
+    }
+    // pre PEP 688 buffer protocol
+    #[cfg(any(Py_3_11, not(Py_LIMITED_API)))]
+    pub unsafe fn __getbuffer__(slf: PyRef<'_, Self>, view: *mut ffi::Py_buffer, flags: c_int) -> PyResult<()> {
+        if view.is_null() {
```

Comment:
judging from opendal this implementation can be simplified a lot: https://github.com/apache/opendal/blob/d001321b0f9834bc1e2e7d463bcfdc3683e968c9/bindings/python/src/utils.rs#L58-L70

Reply:
Yep, first hit in a github-wide search, which I really should have used :facepalm: : https://github.com/search?q=__getbuffer__+language%3ARust+&type=code

Reply:
haha that's how I found it last night 😛